# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

  

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

To add a photo, diagram or document to your file, drop the file into the folder that holds this notebook.  Use the the link button to the right of the  \</\> symbol in the gray part of this cell, the link is just the name of your file.

https://docs.google.com/document/d/1ubxEsp-uSyw88oM9Mf8gvv73tWuy64ZPfMuNx8Ob4T4/edit?usp=sharing

In [ ]:
--SELECT * FROM LaborStatisticsDB.dbo.annual_2016
--SELECT * FROM LaborStatisticsDB.dbo.series
--SELECT * FROM LaborStatisticsDB.dbo.industry
--SELECT * FROM LaborStatisticsDB.dbo.datatype
--SELECT * FROM LaborStatisticsDB.dbo.footnote
SELECT * FROM LaborStatisticsDB.dbo.[period]
--SELECT * FROM LaborStatisticsDB.dbo.seasonal
--SELECT * FROM LaborStatisticsDB.dbo.supersector 
--SELECT * FROM LaborStatisticsDB.dbo.january_2017

2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees? 10</span>

In [32]:
SELECT *
FROM LaborStatisticsDB.dbo.datatype
WHERE data_type_text ='women employees'

(1 row affected)

Total execution time: 00:00:00.111

data_type_code data_type_text 10 WOMEN EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector?</span><span style="font-size: 13.3333px; white-space: pre-wrap;">CES5552211010 and CEU5552211010.</span>

In [33]:
SELECT series_id,data_type_text,industry_name,supersector_name
FROM LaborStatisticsDB.dbo.series AS s
INNER JOIN LaborStatisticsDB.dbo.datatype AS d
ON s.data_type_code = d.data_type_code
INNER JOIN LaborStatisticsDB.dbo.industry AS i  
ON s.industry_code = i.industry_code
INNER JOIN LaborStatisticsDB.dbo.supersector AS ss
ON s.supersector_code = ss.supersector_code
WHERE d.data_type_text = 'women employees'
AND i.industry_name = 'commercial banking'
AND ss.supersector_name = 'financial activities'






(2 rows affected)

Total execution time: 00:00:00.113

series_id data_type_text industry_name supersector_name CES5552211010 WOMEN EMPLOYEES Commercial banking Financial activities CEU5552211010 WOMEN EMPLOYEES Commercial banking Financial activities

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [50]:
SELECT *
FROM LaborStatisticsDB.dbo.annual_2016 AS a
LEFT JOIN LaborStatisticsDB.dbo.series AS s
ON a.series_id = s.series_id

Commands completed successfully.

Total execution time: 00:00:00.087

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [53]:
SELECT * 
FROM LaborStatisticsDB.dbo.series AS s
INNER JOIN LaborStatisticsDB.dbo.datatype AS d
ON s.data_type_code = d.data_type_code

Commands completed successfully.

Total execution time: 00:00:00.087

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [ ]:
SELECT * 
FROM LaborStatisticsDB.dbo.series AS s
INNER JOIN LaborStatisticsDB.dbo.industry AS i
ON s.industry_code = i.industry_code

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please.</span><span style="font-size: 13.3333px; white-space: pre-wrap;">2340612</span>

In [34]:
SELECT ROUND(SUM(a.value), 0) AS Total_Employees_2016
FROM LaborStatisticsDB.dbo.annual_2016 AS a
WHERE a.series_id IN (
    SELECT s.series_id
    FROM LaborStatisticsDB.dbo.series AS s
    WHERE s.series_title = 'All Employees')





(1 row affected)

Total execution time: 00:00:00.116

Total_Employees_2016 2340612

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp;</span> <span style="font-size: 13.3333px; white-space: pre-wrap;">1125490</span>

In [35]:
SELECT ROUND(SUM(a.value),0) AS Total_Women_Employees_2016
FROM LaborStatisticsDB.dbo.annual_2016 AS a
WHERE a.series_id IN (
    SELECT s.series_id
    FROM LaborStatisticsDB.dbo.series AS s
    WHERE s.series_title = 'women employees')

(1 row affected)

Total execution time: 00:00:00.116

Total_Women_Employees_2016 1125490

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;</span> <span style="font-size: 13.3333px; white-space: pre-wrap;">1263650</span>

In [36]:
SELECT ROUND(SUM(a.value), 0) AS Total_Prod_NonSup_Employees_2016
FROM LaborStatisticsDB.dbo.annual_2016 AS a
WHERE a.series_id IN (
    SELECT s.series_id
    FROM LaborStatisticsDB.dbo.series AS s
    WHERE s.series_title = 'production and nonsupervisory employees');

(1 row affected)

Total execution time: 00:00:00.120

Total_Prod_NonSup_Employees_2016 1263650

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries? </span>  36.06

In [37]:
SELECT ROUND(AVG(j.value),2) AS Total_Avg_Weekly_Hrs_Jan2017
FROM LaborStatisticsDB.dbo.january_2017 AS j
WHERE j.series_id IN (
    SELECT s.series_id
    FROM LaborStatisticsDB.dbo.series AS s
    WHERE s.series_title = 'average weekly hours of production and nonsupervisory employees');



(1 row affected)

Total execution time: 00:00:00.139

Total_Avg_Weekly_Hrs_Jan2017 36.06

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please. </span> <span style="font-size: 13.3333px; white-space: pre-wrap;">1838753220</span>

In [38]:
SELECT ROUND(SUM(j.value),2) AS Total_Weekly_Payroll_2017
FROM LaborStatisticsDB.dbo.january_2017 AS j
WHERE j.series_id IN (
    SELECT s.series_id
    FROM LaborStatisticsDB.dbo.series AS s
    WHERE s.series_title = 'AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES');




(1 row affected)

Total execution time: 00:00:00.127

Total_Weekly_Payroll_2017 1838753220

6. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest? </span> Motor vehicle power train components industry was the average weekly hours worked by production and nonsupervisory employees the highest and  Fitness and recreational sports centers was the lowest.

In [39]:
SELECT TOP 1 i.industry_name, AVG(j17.value) AS Avg_weekly_hours
FROM LaborStatisticsDB.dbo.datatype AS d
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON d.data_type_code = s.data_type_code
INNER JOIN LaborStatisticsDB.dbo.january_2017 AS j17
ON s.series_id = j17.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i 
ON  i.industry_code = s.industry_code 
WHERE d.data_type_code ='7'
GROUP BY i.industry_name
ORDER BY Avg_weekly_hours DESC

SELECT  TOP 1 i.industry_name, AVG(j17.value) AS Avg_weekly_hours
FROM LaborStatisticsDB.dbo.datatype AS d
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON d.data_type_code = s.data_type_code
INNER JOIN LaborStatisticsDB.dbo.january_2017 AS j17
ON s.series_id = j17.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i 
ON  i.industry_code = s.industry_code 
WHERE d.data_type_code ='7'
GROUP BY i.industry_name
ORDER BY Avg_weekly_hours 


(1 row affected)

(1 row affected)

Total execution time: 00:00:00.196

industry_name Avg_weekly_hours Motor vehicle power train components 49.6

industry_name Avg_weekly_hours Fitness and recreational sports centers 16.85

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest?</span>Private sector was the average weekly hours worked by production and nonsupervisory employees the highest and  Coin-operated laundries and drycleaners was the lowest.

In [40]:
SELECT  TOP 1 i.industry_name, SUM(j17.value) AS Total_weekly_payroll_PSE_Jan2017
FROM LaborStatisticsDB.dbo.datatype AS d
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON d.data_type_code = s.data_type_code
INNER JOIN LaborStatisticsDB.dbo.january_2017 AS j17
ON s.series_id = j17.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i 
ON  i.industry_code = s.industry_code 
WHERE d.data_type_code ='82'
GROUP BY i.industry_name
ORDER BY Total_weekly_payroll_PSE_Jan2017 DESC

SELECT TOP 1 i.industry_name, SUM(j17.value) AS Total_weekly_payroll_PSE_Jan2017
FROM LaborStatisticsDB.dbo.datatype AS d
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON d.data_type_code = s.data_type_code
INNER JOIN LaborStatisticsDB.dbo.january_2017 AS j17
ON s.series_id = j17.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i 
ON  i.industry_code = s.industry_code 
WHERE d.data_type_code ='82'
GROUP BY i.industry_name
ORDER BY Total_weekly_payroll_PSE_Jan2017 



(1 row affected)

(1 row affected)

Total execution time: 00:00:00.172

industry_name Total_weekly_payroll_PSE_Jan2017 Total private 295944946

industry_name Total_weekly_payroll_PSE_Jan2017 Coin-operated laundries and drycleaners 40448

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [46]:
SELECT j.series_id,i.industry_code,i.industry_name,j.value
FROM january_2017 AS j
INNER JOIN series AS s
ON j.series_id = s.series_id
INNER JOIN industry AS i
ON s.industry_code = i.industry_code
WHERE data_type_code = 82 AND j.value > 
      (SELECT AVG(a.value) 
      FROM LaborStatisticsDB.dbo.annual_2016 AS a
      WHERE series_id like '%82')


Commands completed successfully.

Total execution time: 00:00:00.088

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [31]:
SELECT a.year,a.period,(ROUND(AVG(a.value),2)) AS Avg_Earnings
FROM LaborStatisticsDB.dbo.annual_2016 AS a
WHERE a.series_id like '%30'--last 2 digits of series_id and datatype are linked
GROUP BY a.year,a.period
UNION 
SELECT j.year,j.period,(ROUND(AVG(j.value),2))AS Avg_Earnings
FROM LaborStatisticsDB.dbo.january_2017 AS j
WHERE j.series_id like '%30'
GROUP BY j.year,j.period


(2 rows affected)

Total execution time: 00:00:00.251

year period Avg_Earnings 2016 M13 797.2 2017 M01 808.53

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

During Jan 2017,production and non supervisory employed fare better because the avg earnings was 808.53 which is better than  2016 avg earnings .

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better?</span>

Based on the analysis,private sector fare better because the total weekly payroll is highest which is 295944946.

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</span>

Values column was little confusing,it would have been better if data was clearly divided into total number of employees, total/Avg hours,Total/Avg earnings.There are some unnecessary data(tables /columns) which is not required for the analysis and it can be removed during data cleaning like footnote etc.